functions.ts

```
@@ -1,12 +1,48 @@
+import { logging } from "./logging";
+
+export const createNanoEvents = () => ({
+  emit<K extends keyof SpectodaJsEventMap>(event: K, ...args: SpectodaJsEventMap[K]) {
+    const callbacks = this.events[event] || [];
+    for (let i = 0, length = callbacks.length; i < length; i++) {
+      callbacks[i](...args);
+    }
+  },
+  events: {},
+  on<K extends keyof SpectodaJsEventMap>(event: K, cb: (props: SpectodaJsEventMap[K]) => void) {
+    this.events[event]?.push(cb) || (this.events[event] = [cb]);
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)